packages/app/src/cli/models/extensions/extensions.ts

```diff
@@ -116,7 +118,7 @@ export class ExtensionInstance<TConfiguration extends BaseConfigContents = BaseC
     this.idEnvironmentVariableName = `SHOPIFY_${string.constantize(path.basename(this.directory))}_ID`
   }
 
-  async build(stderr: Writable, stdout: Writable, app: AppInterface) {
+  async build(stdout: Writable, stderr: Writable, app: AppInterface) {
```

Comment:
I added this here but I'm not actually using it because I wasn't sure about build vs dev yet.
But didn't thought that different extensions ui could have different build processes, that makes sense 🤔 

If it's ok with you, I can improve this part on a future PR, adding a `buildForDev` method or similar with a default behaviour

Reply:
That works.  

The different build behaviour is around inserting the code to register each extension point. See [this section here](https://docs.google.com/document/d/1rlv3DcgGZi4_Hf_Ekd14DnYWJ1rjsfZlMBVC80gFWjc/edit#heading=h.45n3s59gtwio).  The files in src/ for old extensions will register their extension points, but the new UI extension (which everything will migrate to) will have this code automatically injected by esbuild.